src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.AddTransactionCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Amount;
+import seedu.address.model.person.Description;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.UniquePersonList;
+import seedu.address.model.transaction.Transaction;
+
+/**
+ * Parses input arguments and creates a new AddTransactionCommand object
+ */
+public class AddTransactionCommandParser implements Parser<AddTransactionCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddTransactionCommand
+     * and returns an AddTransactionCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddTransactionCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_NAME);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_NAME)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTransactionCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            Name payerName = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME)).get();
+            Amount amount = ParserUtil.parseAmount(argMultimap.getValue(PREFIX_AMOUNT)).get();
+            Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION)).get();
+            Name payeeName= ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME)).get();
```

Comment:
This gets only one name of the payee. You should check how the code deals with tags (multi-optional input)